src/zql/ast-to-ivm/pipeline-builder.ts

```diff
@@ -297,53 +292,73 @@ function makeKeyFunction(columns: string[]) {
 // We're well-typed in the query builder so once we're down here
 // we can assume that the operator is valid.
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
-function getOperator(op: SimpleOperator): (l: any, r: any) => boolean {
+export function getOperator(condition: SimpleCondition): (lhs: any) => boolean {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const rhs = condition.value.value as any;
```

Comment:
I changed these to capture the RHS so that the compilation of the pattern to a regexp is done at the time when we build the pipeline.

Maybe this needs to be changed if we do placeholders but we can deal with that later.